### PR TITLE
refactor: HTTP error codes GET/users/{user_id}

### LINKED ISSUE #623: Fix: refactor HTTP error codes GET/users/{user_id}
## Description
As a GSoC student,
I need the correct HTTP error response code to be returned accordingly by Mentorship System API in relation to GET /users/{user_id},
so that I can represent the error message accordingly to both Mentorship System and Bridge In Tech users.

## Mocks
NA
### Update [Required]
- [ ] 200: OK needs to be returned instead of 201: Created if the process is successful
- [ ] 403: Forbidden needs to be returned instead of 400: Bad Request if the user_id who sends the request is not valid the operation.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by satya7289**
### Description

Refactor HTTP status code from 400->403(Bad Request->Forbidden) and 201->200(Created->Ok).
And add corresponding Test cases.

Fixes #623 

### Type of Change:

- Code
- Quality Assurance
- User Interface

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

Screenshot from 2020-06-06 20-33-57

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] Update Swagger documentation and the exported file at /docs folder

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by SanketDG — DISMISSED** [review]
LGTM can be merged @anitab-org/mentorship-backend-maintainers

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@satya7289 this looks great :) Thank you for this contribution 🎉 

As an enhancement and part of our guidelines, could you please add a unit test per each error code you fixed to make sure this is working as intended? So 2 small tests to test that the correct code is returned. You can check other tests to get an example.

Let me know if you need any help!

File: app/api/resources/user.py
Comment (satya7289) [inline app/api/resources/user.py]: FORBIDDEN are not raising if we are using  `users/<int:user_id>`. On routing `localhost:5000/users/abc` then this error pop up `The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.`
So either we should remove `users/<int:user_id>` to `users/<user_id>` or remove `FORBIDDEN status`.
@isabelcosta

Comment (isabelcosta) [reply app/api/resources/user.py]: Very interesting 🤔 I am not sure what the "best" route is here?!
I wonder if this check is even needed... What would you suggest from the 2 options you suggested @satya7289 ?

Also... @ramitsawhney27 @m-murad

Comment (satya7289) [reply app/api/resources/user.py]: I would suggest remove the `FORBIDDEN` status. @isabelcosta

File: tests/users/test_api_user.py
Comment (isabelcosta) [inline tests/users/test_api_user.py]:
```suggestion
            f"/users/{self.verified_user.id}", follow_redirects=True, headers=auth_header
```
the API is described as `GET /users/<user_id>` instead of as query parameter `GET /users?user_id=<user_id>`

**Review by isabelcosta — CHANGES_REQUESTED** [review]
This looks great now @satya7289 
I left a change request because of the use of the API, aside from that could you also add a test, for when you give an invalid user id, the response returned?

**Comment by satya7289** [comment]
Any more changes requires..!
@isabelcosta @mtreacy002

**Comment by mtreacy002** [comment]
Thanks @satya7289 for your contribution. Sorry for the late reply, I'm a bit tight in time so will pass the request to @anitab-org/coding-team or @anitab-org/bridgeintech-maintainers to help review this PR.

File: tests/users/test_api_user.py
Comment (isabelcosta) [inline tests/users/test_api_user.py]: can you use HTTP status code from HTTPStatus module like HTTPStatus.OK? and do the same for the next line 57 and 65

File: tests/users/test_api_user.py
Comment (isabelcosta) [inline tests/users/test_api_user.py]:
```suggestion
class TestnGetOtherUserApi(BaseTestCase):
```
To be more explicit about the tests that you are running here, which is about get other user profile API

File: tests/users/test_api_user.py
Comment (isabelcosta) [inline tests/users/test_api_user.py]: Can you rename this file to `.../test_api_get_other_user.py`?

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@satya7289 this is almost done :) I suggested some minor changes.

File: tests/users/test_api_get_other_user.py.py
Comment (isabelcosta) [inline tests/users/test_api_get_other_user.py.py]: this test is the exact same test as the above right @satya7289 ?

**Review by isabelcosta — CHANGES_REQUESTED** [review]

File: tests/users/test_api_get_other_user.py.py
Comment (satya7289) [reply tests/users/test_api_get_other_user.py.py]: Yes, almost but in this test case we are testing ok error code for invalid user.

Comment (isabelcosta) [reply tests/users/test_api_get_other_user.py.py]: Ok, at first I thought the code was the exact same, but now I understand the difference. Okay so I would suggest something else here. I will add a comment in separate.

File: tests/users/test_api_get_other_user.py.py
Comment (isabelcosta) [inline tests/users/test_api_get_other_user.py.py]: I think this is misleading and perhaps this is not what you want to test. You are saying here that you expect to see `other_user` being returned, but you are calling the endpoint with `verified_user`, so it's natural that the responses are not the same.  So you are not actually expecting that. You are asserting something that you know will not be the case. For example, you could have this line 59 (calling `self.other_user`) under 51 in the other test, and use it as a second assertion. I am not sure if this is clearer?

I would rather suggest you make this test about an unauthenticated user, so for example, if you don't provide the auth header in line 56, you don't get a successful response because the access token is missing. However this is out of the scope of this PR, you can remove this test, since you are calling the same endpoint with the same data, just making a different assertion.

**Review by isabelcosta — CHANGES_REQUESTED** [review]
@satya7289 I left a change request, let me know what you think.

File: tests/users/test_api_get_other_user.py.py
Comment (satya7289) [reply tests/users/test_api_get_other_user.py.py]: Ok, I'll remove this test case.

Comment (satya7289) [reply tests/users/test_api_get_other_user.py.py]: Any other changes needs to do other then this, please let me know.

File: tests/users/test_api_get_other_user.py.py
Comment (isabelcosta) [inline tests/users/test_api_get_other_user.py.py]: the name of the file ends with ".py.py", can you please edit it?

**Review by isabelcosta — CHANGES_REQUESTED** [review]

**Review by isabelcosta — APPROVED** [review]
Great work @satya7289 👏 🎉

**Review by SanketDG — COMMENTED** [review]
Looks good, @satya7289 can you just re-push once to trigger the new test runs?

Also, feel free to squash any commits as required and also make sure to follow the commit message guidelines

**Comment by isabelcosta** [comment]
@SanketDG @satya7289 no need to squash because I can then squash it on my end :)

**Comment by isabelcosta** [comment]
Ah, I get now. @satya7289 can you update your branch with latest develop code and push it

**Comment by satya7289** [comment]
> Ah, I get now. @satya7289 can you update your branch with latest develop code and push it

Done @isabelcosta

**Comment by isabelcosta** [comment]
All good, you managed to increase the test coverage 👏 @satya7289 
@SanketDG can you please approve this if you find no problem in this PR?

**Review by SanketDG — APPROVED** [review]
I have no idea how this slipped. LGTM!

**Comment by isabelcosta** [comment]
No problem @SanketDG 🙈 thank you so much for helping out with the reviews 🙌

**Review by Abdelaty — APPROVED** [review]
LGTM @satya7289 💯   Thank you, you can merge commits to one commit if it is possible.

**Comment by gaurivn** [comment]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done or Not Done

2. All possible responses (positive and negative tests) were tested as below:

  * _Before PR637_
    _Screenshot_:
beforePR637

    _Expected Result_:  Error codes 201 and 400
    _Actual Result_: Same as expected

  * _After PR637_
    _Screenshot_:  
afterPR637

    _Expected Result_:  200: OK is returned instead of 201: Created if the process is successful
    _Actual Result_:  Same as above
    ...  

3. OS Version: Ubuntu 18.04.4 LTS

**Comment by isabelcosta** [comment]
Thank you so much @gaurivn !

**Comment by gaurivn** [comment]
Welcome, @isabelcosta I'm not sure why I'm unable to see 403 though, have I done it wrong, I will try to do a code review too

**Comment by isabelcosta** [comment]
@gaurivn you don't need to see the 403, this issue was discussed in this comment thread: https://github.com/anitab-org/mentorship-backend/pull/637#discussion_r438565233

**Comment by gaurivn** [comment]
Okay I got it now, thank you

**Comment by rpattath** [comment]
@gaurivn would you mind adding these tests to [TestQuality](https://anitab-org.testquality.com/signup). Ping me know if you need help.

**Comment by gaurivn** [comment]
> @gaurivn would you mind adding these tests to [TestQuality](https://anitab-org.testquality.com/signup). Ping me know if you need help.

Sure I will try adding the tests

**Pull request merged**
